src/openms/source/ANALYSIS/OPENSWATH/OpenSwathScoring.cpp

```diff
@@ -86,6 +89,33 @@ namespace OpenMS
     std::vector<double> normalized_library_intensity;
     getNormalized_library_intensities_(transitions, normalized_library_intensity);
 
+    // If add_up_spectra_ is -1, we automatically compute the amount of spectra to add by setting add_up_spectra_ to the ceiling value of N percent of the retention time peak width
+    if (add_up_spectra_ == -1)
+    {
+        double leftWidth = imrmfeature->getMetaValue("leftWidth");
+        double rightWidth = imrmfeature->getMetaValue("rightWidth");
+        double peakWidth = rightWidth - leftWidth;
+        add_up_spectra_ = std::ceil(peakWidth * use_percent_peak_width_);
+        
+        // Ensure add_up_spectra_ is odd
+        if (static_cast<int>(add_up_spectra_) % 2 == 0) {
```

Comment:
I would change the value such that we don't need to have these kind of checks.
e.g. if  +-add_up_spectra are extracted, we would not need such a check and "hack" to make it a valid ordinal range to extract

Reply:
I'm not sure if this is what you meant, or the best way to do this, but I now do:

```
n_merge_spectra = std::max(1, static_cast<int>(std::ceil(peakWidth * merge_spectra_by_peak_width_fraction_)) | 1);
```

- Use 1 if the resulting expression on the right is 0 or less
- if the calculation of ceiling peak width * N frac is even, use bitwise OR operation to ensure it's odd.